numpy/lib/arraysetops.py

```diff
@@ -481,6 +485,59 @@ def in1d(ar1, ar2, assume_unique=False, invert=False):
     else:
         return ret[rev_idx]
 
+
+def isin(elements, test_elements, **kwargs):
+    """
+    Test whether each element of an array is also present in a second array.
+    Returns a boolean array of the same shape as `elements` that is True
+    where an element of `elements` is in `test_elements` and False otherwise.
+
+    Parameters
+    ----------
+    elements : array_like
+        Input array.
+    test_elements : iterable
+        The values against which to test each value of `elements`.
+        This argument is flattened if it is an array or array_like.
+    **kwargs: Keyword arguments passed to `in1d`.
+
+    Returns
+    -------
+    isin : ndarray, bool
+        Has the same shape as `elements`. The values `elements[isin]`
+        are in `test_elements`.
+
+    See Also
+    --------
+    in1d                  : Flattened version of this function.
+    numpy.lib.arraysetops : Module with a number of other functions for
+                            performing set operations on arrays.
+    Notes
+    -----
+    .. versionadded:: 1.13.0
+
+    Examples
+    --------
+    >>> elements = np.array([[0, 2], [4, 6]])
+    >>> test_elements = {1, 2, 4, 8}
+    >>> mask = np.isin(elements, test_elements)
+    >>> mask
+    array([[ False,  True],
+           [ True,  False]], dtype=bool)
+    >>> elements[mask]
+    array([2, 4])
+    >>> mask = np.isin(elements, test_elements, invert=True)
+    >>> mask
+    array([[ True, False],
+           [ False, True]], dtype=bool)
+    >>> elements[mask]
+    array([0, 6])"""
+    elements = np.array(elements)
+    if not hasattr(test_elements, '__array__'):
+        test_elements = np.array([i for i in test_elements])
```

Comment:
Instead of the list comprehension, just cast to `list`, e..g, `np.array(list(test_elements))`.